src/test/kotlin/uk/gov/justice/digital/hmpps/approvedpremisesapi/unit/service/cas2/PersonUtilsTest.kt

```diff
@@ -0,0 +1,79 @@
+package uk.gov.justice.digital.hmpps.approvedpremisesapi.unit.service.cas2
+
+import io.mockk.every
+import org.assertj.core.api.Assertions
+import org.junit.jupiter.api.Test
+import org.junit.jupiter.api.assertThrows
+import org.springframework.http.HttpMethod
+import org.springframework.http.HttpStatus
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.client.ClientResult
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.factory.InmateDetailFactory
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.factory.OffenderDetailsSummaryFactory
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.model.PersonInfoResult
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.problem.NotFoundProblem
+import uk.gov.justice.digital.hmpps.approvedpremisesapi.util.cas2.getFullInfoForPersonOrThrow
+
+class PersonUtilsTest : OffenderServiceTest() {
```

Comment:
ah sorry I didn't see this was extending the other test class. This is a tricky one. I'd probably have opted for not extending the tests and doing some setup in this file, just so it's easier for readers to not have to click through files to see e.g. where `mockOffenderDetailsDataSource` is defined. We also don't tend to have many other test files extending test files elsewhere on the project (other than the `BaseIntegrationTest`).

However, this defining of a function on the `OffenderService` in a separate utils file is a bit new to me too, so it's an unusual scenario!

I'd still probably opt for the former to help with readability, but could be swayed.

Any thoughts @Harriethw ?

Reply:
Yes I don't know why we have two functions that extend the CAS2 `OffenderService` in the `PersonUtil` module either really.  If they were in the same module we could remove `PersonUtils` and include the tests in `OffenderServiceTest` and therefore no requirement for a dedicated `PersonUtilsTest` any more.

Reply:
Looking at the git history, I wonder if `PersonUtil` was a legacy of splitting off from CAS1, or just a first attempt at breaking out some things from the overly-long Offender Service... if it's causing confusion, happy for the functions to be brought back into the `OffenderService` and then just tested within `OffenderServiceTest` .

I think regardless of whether we remove `PersonUtil` we could test it without inheritance - as George suggests, by bringing in the setup to the `PersonUtilsTest` file rather than inheriting.

Reply:
PersonUtils.kt has gone in the most recent commit and the extension functions in it have now been moved to OffenderService.kt.

All tests are now in OffenderServiceTest.kt.

Nice and tidy.